subprojects/instant-execution/src/integTest/groovy/org/gradle/instantexecution/InstantExecutionTaskExecutionIntegrationTest.groovy

```diff
@@ -50,4 +52,115 @@ class InstantExecutionTaskExecutionIntegrationTest extends AbstractInstantExecut
         result.assertTaskSkipped(":always")
         result.assertTasksNotSkipped(":never")
     }
+
+    def "honors task timeout"() {
```

Comment:
For these kinds of changes, I think it would be better to go and improve the existing tests for timeout, enabled, etc and make sure they run the build a couple of times, rather than adding duplicate test coverage here.

Reply:
👍 Done